src/main/io/ledstrip.h

```diff
@@ -21,11 +21,50 @@
 #define LED_CONFIGURABLE_COLOR_COUNT   16
 #define LED_MODE_COUNT                  6
 #define LED_DIRECTION_COUNT             6
-#define LED_FUNCTION_COUNT             10
-#define LED_SPECIAL_COLOR_COUNT         8
+#define LED_BASEFUNCTION_COUNT          7
+#define LED_OVERLAY_COUNT               6
+#define LED_SPECIAL_COLOR_COUNT        11
+
+#define LED_POS_OFFSET                  0
+#define LED_FUNCTION_OFFSET             8
+#define LED_OVERLAY_OFFSET             12
+#define LED_COLOR_OFFSET               18
+#define LED_DIRECTION_OFFSET           22
+#define LED_PARAMS_OFFSET              28
+
+#define LED_POS_BITCNT                  8
+#define LED_FUNCTION_BITCNT             4
+#define LED_OVERLAY_BITCNT              6
+#define LED_COLOR_BITCNT                4
+#define LED_DIRECTION_BITCNT            6
+#define LED_PARAMS_BITCNT               4
+
+#define LED_FLAG_OVERLAY_MASK ((1 << LED_OVERLAY_BITCNT) - 1)
+#define LED_FLAG_DIRECTION_MASK ((1 << LED_DIRECTION_BITCNT) - 1)
+
+#define LED_MOV_POS(pos) ((pos) << LED_POS_OFFSET)
+#define LED_MOV_FUNCTION(func) ((func) << LED_FUNCTION_OFFSET)
+#define LED_MOV_OVERLAY(overlay) ((overlay) << LED_OVERLAY_OFFSET)
+#define LED_MOV_COLOR(colorId) ((colorId) << LED_COLOR_OFFSET)
+#define LED_MOV_DIRECTION(direction) ((direction) << LED_DIRECTION_OFFSET)
+#define LED_MOV_PARAMS(param) ((param) << LED_PARAMS_OFFSET)
+
+#define MASK(len) ((1 << (len)) - 1)
```

Comment:
This is too generic name for header file. It will clash with something eventually ..